smart-contracts/advanced/multicall.md

```diff
@@ -0,0 +1,55 @@
+---
+description: >-
+  Multicall allows you to aggregate multiple contract reads into a single JSON-RPC request,
+  and execute multiple state-changing calls in a single transaction on the FVM.
+---
+
+# Multicall
+
+Multicall is a powerful tool that offers several advantages for interacting with smart contracts on the Filecoin Virtual Machine (FVM). 
+
+## Contract address
+
+Multicall has the same, precomputed address for all of the networks it is deployed on. 
+| Name             | Address                                      | Mainnet | Calibration |
+| ---------------- | -------------------------------------------- | ------- | ----------- |
+| [Multicall - Mainnet](https://filfox.info/en/address/0xcA11bde05977b3631167028862bE2a173976CA11?t=3) | `0xcA11bde05977b3631167028862bE2a173976CA11` | ✔️      | ❌          |
+| [Multicall - Calibration](https://calibration.filscan.io/en/address/0xcA11bde05977b3631167028862bE2a173976CA11/) | `0xcA11bde05977b3631167028862bE2a173976CA11` | ❌      | ✔️          |
+
+## Usage
+
+Batching contract reads, one of the most common use cases, allows a single `eth_call` JSON RPC request to return the results of multiple contract function calls. It has many benefits:
+
+1. **Reduced JSON RPC Requests**: Multicall reduces the number of separate JSON RPC requests that need to be sent. This is particularly useful when using remote nodes like Infura. By aggregating multiple contract reads into a single JSON-RPC request, Multicall (1) reduces RPC usage and therefore costs, and (2) reduces the number of round trips between the client and the node, which can significantly improve performance.
+
+2. **Consistent Data from the Same Block**: Multicall guarantees that all values returned are from the same block. This ensures data consistency and reliability, as all the read operations are performed on the same state of the blockchain.
+
+3. **Detection of Stale Data**: Multicall enables the block number or timestamp to be returned with the read data. This feature helps in detecting stale data, as developers can compare the block number or timestamp with the current state of the blockchain to ensure the data is up-to-date.
+
+Many libraries and tools such as ethers-rs, viem, and ape have native Multicall3 integration.
```

Comment:
We should add some links to ethers-rs, view, and also the link to Multicall3 examples.

Reply:
Linked; implemented a multicall3 example since I didn't see any from their docs